File: processing/pong/src/pong.c
#include <stdio.h>
#include <time.h>
#include <string.h>

#define BUF_LEN 256

int main(void) {
    char buf[BUF_LEN] = {0};
    time_t rawtime = time(NULL);
    struct tm *ptm = gmtime(&rawtime);
    strftime(buf, BUF_LEN, "%FT%X.000Z", ptm);

    printf("{\"msg\": \"pong\", \"process\": \"c\", \"created\":\"%s\"}", buf);

    return 0;
}

File: processing/pong/pong.pl
#!/usr/bin/env perl

use POSIX qw(strftime);
use Time::HiRes qw/gettimeofday/;

sub iso8601
{
  my ($s, $f) = split (/\./, gettimeofday);
  strftime ('%Y-%m-%dT%H:%M:%S.'.$f.'Z', gmtime ($s))
}

print '{"msg": "pong", "process": "perl", "created": "' . iso8601 . '"}';
